### DISCUSSION TITLE
Create Printable Reports

### DISCUSSION THREAD

**Discussion by mis-uep** [phalcon/cphalcon#15609] [Q&A]
Hi! How can I create a printable Reports that can be viewed in print preview first?
Thanks!

**Comment by Jeckerson** [comment]
Hello. What kind of reports?

**Reply by mis-uep** [reply]
Hi, or a way maybe to create a pdf file?

**Reply by wurst-hans** [reply]
This seems not to be related to Phalcon? Use any (free) PHP library for generating PDF files like TCPDF, MPDF and so on. Many of them allow to generate a PDF using HTML code.

**Reply by niden** [reply]
Phalcon will not create any PDF files for you. The functionality is not built into the framework. 

You will need a different library to tie it with Phalcon and then produce your documents. Along with @wurst-hans 's suggestions, you could also use https://github.com/mikehaertl/phpwkhtmltopdf which wraps `wkhtmltopdf` in a library that you can use from your code. You can then create a simple view of what you want your PDF to be, feed it to that library and you have your PDF file.

**Reply by mis-uep** [reply]
Thanks for the suggestions!
